selfhost/typechecker.jakt

```diff
@@ -4112,6 +4112,17 @@ struct Typechecker {
             .error("‘return’ is not allowed inside ‘defer’", span)
         }
         if not expr.has_value() {
+            if .current_function_id.has_value() {
```

Comment:
Nice fix.

Can you also add a test so that we can make sure this continues to work in the future?

Reply:
I just realized that this breaks the compiler. I'm trying to find what condition I missed.

Reply:
I added the test in the last force push.